### PR TITLE
Add a rate parameter to Reaction and FVReaction

### LINKED ISSUE #16187: Rate parameter for `Reaction` and `FVReaction` kernels
## Reason
The existing reaction kernels are limited to the special case of a unity rate coefficient. Non-unity coefficients are needed for basically any real physics, e.g. decay of a radioactive material.

## Design
The `Reaction` and `FVReaction` kernels should accept some rate parameter. Perhaps for more general applications, this rate parameter should be a material property or a coupled variable, but a simple scalar value will serve for my current use-case.

## Impact
New feature.

### PR DESCRIPTION
**Pull request by smharper**
Closes #16187

### CODE REVIEW

File: framework/include/fvkernels/FVReaction.h [hunk 1]
```diff
@@ -19,4 +19,6 @@ class FVReaction : public FVElementalKernel
 
 protected:
   ADReal computeQpResidual() override;
+
+  Real _rate;
```
Comment (rwcarlsen) [inline framework/include/fvkernels/FVReaction.h:23]: `const Real &`.

File: framework/include/kernels/Reaction.h [hunk 2]
```diff
@@ -27,4 +27,6 @@ class Reaction : public Kernel
 protected:
   virtual Real computeQpResidual() override;
   virtual Real computeQpJacobian() override;
+
+  Real _rate;
```
Comment (rwcarlsen) [inline framework/include/kernels/Reaction.h:31]: `const Real &`

File: test/tests/fvkernels/fv_coupled_var/coupled.i [hunk 3]
```diff
@@ -49,6 +50,7 @@
   [rxn]
     type = FVReaction
     variable = v
+    rate = 2.0
```
Comment (rwcarlsen) [inline test/tests/fvkernels/fv_coupled_var/coupled.i:53]: Did you manually verify the results that you are regolding here to ensure they are correct?

**Review by rwcarlsen — CHANGES_REQUESTED** [review]
Looks good.  Just a couple small things.

File: test/tests/fvkernels/fv_coupled_var/coupled.i (hunk 3 above)
Comment (smharper) [reply test/tests/fvkernels/fv_coupled_var/coupled.i:53]: I did. This test allows you to double check that the FV and FE results are consistent (and they are). Separately, I've found that the results match the expected pen-and-paper value for a source-decay problem.

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**